core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/deletion/actions/WsmDeletionAction.scala

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.dsde.rawls.monitor.workspace.runners.deletion.actions
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.StatusCodes
+import bio.terra.workspace.client.ApiException
+import bio.terra.workspace.model.JobReport.StatusEnum
+import bio.terra.workspace.model.JobResult
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.workspacemanager.WorkspaceManagerDAO
+import org.broadinstitute.dsde.rawls.model.{RawlsRequestContext, Workspace}
+import org.broadinstitute.dsde.rawls.monitor.workspace.runners.deletion.actions.DeletionAction.when500
+import org.broadinstitute.dsde.rawls.util.Retry
+import org.broadinstitute.dsde.rawls.workspace.WorkspaceManagerOperationFailureException
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future, blocking}
+import scala.util.{Failure, Success, Try}
+
+class WsmDeletionAction(workspaceManagerDao: WorkspaceManagerDAO,
```

Comment:
Much of this code can be made common w/the existing stuff in the MultiCloudWorkspaceService. I'm limiting the scope here but will ticket out that remediation.